RecoLocalCalo/HcalRecProducers/src/HBHEPlan1Combiner.cc

```diff
@@ -0,0 +1,221 @@
+// -*- C++ -*-
+//
+// Package:    RecoLocalCalo/HcalRecProducers
+// Class:      HBHEPlan1Combiner
+// 
+/**\class HBHEPlan1Combiner HBHEPlan1Combiner.cc RecoLocalCalo/HcalRecProducers/plugins/HBHEPlan1Combiner.cc
+
+ Description: rechit combiner module for the "Plan 1" scenario
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Igor Volobouev
+//         Created:  Sun, 29 Jan 2017 16:05:31 GMT
+//
+//
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "RecoLocalCalo/HcalRecAlgos/interface/HBHERecHitAuxSetter.h"
+#include "RecoLocalCalo/HcalRecAlgos/interface/CaloRecHitAuxSetter.h"
+#include "RecoLocalCalo/HcalRecAlgos/interface/parsePlan1RechitCombiner.h"
+
+#include "Geometry/Records/interface/HcalRecNumberingRecord.h"
+#include "Geometry/CaloTopology/interface/HcalTopology.h"
+
+//
+// class declaration
+//
+class HBHEPlan1Combiner : public edm::stream::EDProducer<>
+{
+public:
+    explicit HBHEPlan1Combiner(const edm::ParameterSet&);
+    ~HBHEPlan1Combiner();
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+    virtual void produce(edm::Event&, const edm::EventSetup&) override;
+
+    //virtual void beginStream(edm::StreamID) override;
+    //virtual void endStream() override;
+    //virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+    //virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+    //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+    //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant